sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -53,47 +60,253 @@ def _is_ipython(self):
         try:
             import IPython
+
             ipy = IPython.get_ipython()
             if ipy is None:
                 return False
         except ImportError:
             return False
         return True
 
-    def create_experiment(self, exp_object, namespace=None):
+    def create_experiment(
+        self, exp_object, namespace=utils.get_default_target_namespace()
+    ):
         """Create the Katib Experiment.
 
         :param exp_object: Experiment object.
         :param namespace: Experiment namespace.
         If the namespace is None, it takes namespace from the Experiment or "default".
-
-        :return: Created Experiment.
-        :rtype: dict
         """
 
-        if namespace is None:
-            namespace = utils.set_katib_namespace(exp_object)
         try:
-            outputs = self.api_instance.create_namespaced_custom_object(
+            self.api_instance.create_namespaced_custom_object(
                 constants.KUBEFLOW_GROUP,
                 constants.KATIB_VERSION,
                 namespace,
                 constants.EXPERIMENT_PLURAL,
-                exp_object)
+                exp_object,
+            )
         except client.rest.ApiException as e:
             raise RuntimeError(
                 "Exception when calling CustomObjectsApi->create_namespaced_custom_object:\
-         %s\n" % e)
+         %s\n"
+                % e
+            )
+
+        # TODO (andreyvelich): Use proper logger.
+        print("Experiment {} has been created".format(exp_object.metadata.name))
 
         if self._is_ipython():
             if self.in_cluster:
                 import IPython
-                IPython.display.display(IPython.display.HTML(
-                    'Katib Experiment {} '
-                    'link <a href="/_/katib/#/katib/hp_monitor/{}/{}" target="_blank">here</a>'.format(
-                        exp_object.metadata.name, namespace, exp_object.metadata.name)
-                ))
-        return outputs
+
+                IPython.display.display(
+                    IPython.display.HTML(
+                        "Katib Experiment {} "
+                        'link <a href="/_/katib/#/katib/hp_monitor/{}/{}" target="_blank">here</a>'.format(
+                            exp_object.metadata.name,
+                            namespace,
+                            exp_object.metadata.name,
+                        )
+                    )
+                )
+
+    def tune(
+        self,
+        # TODO (andreyvelich): How to be consistent with other APIs (name) ?
+        name: str,
+        objective: Callable,
+        parameters: Dict[str, Any],
+        base_image: str = constants.BASE_IMAGE_TENSORFLOW,
+        namespace: str = utils.get_default_target_namespace(),
+        algorithm_name: str = "random",
+        objective_metric_name: str = None,
+        additional_metric_names: List[str] = [],
+        objective_type: str = "maximize",
```

Comment:
maybe we could use "maximize" and "minimize" as an `Enum` class or `Literal`

Reply:
I was also thinking about it. 
Is there a way to automatically generate enums for our [Python Models](https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/models/v1beta1_objective_spec.py#L40) ?
In our APIs `type` param is enum: https://github.com/kubeflow/katib/blob/master/pkg/apis/controller/common/v1beta1/common_types.go#L96

Reply:
By "automatically" means that automatically generate python Enum class from the golang file or openapi.json ?

Reply:
> By "automatically" means that automatically generate python Enum class from the golang file or openapi.json ?

Exactly. I am not sure if `kube-openapi` supports enum generation. Check this: https://github.com/kubernetes/enhancements/issues/2887.
Maybe we should update `kube-openapi` version for this feature.

Reply:
I have no many experience with it sadly. How about make an issue with this feature, and merge this PR with hardcoded for now ?

Reply:
Sounds good @anencore94 !